packages/web-react/src/components/FileUploader/FileUploaderInput.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import classNames from 'classnames';
+import { SpiritFileUploaderInputProps } from '../../types';
+import { useStyleProps, useValidationText } from '../../hooks';
+import { DEFAULT_FILE_QUEUE_LIMIT, DEFAULT_FILE_SIZE_LIMIT } from './constants';
+import { useFileUploaderStyleProps } from './useFileUploaderStyleProps';
+import { useFileUploaderInput } from './useFileUploaderInput';
+import { Icon } from '../Icon';
+
+const FileUploaderInput = (props: SpiritFileUploaderInputProps) => {
+  const {
+    accept,
+    dropZoneRef,
+    helperText,
+    id,
+    inputRef,
+    isDisabled,
+    isMultiple,
+    isRequired,
+    label,
+    labelText,
+    linkText,
+    maxFileSize = DEFAULT_FILE_SIZE_LIMIT,
+    maxUploadedFiles = DEFAULT_FILE_QUEUE_LIMIT,
+    onError,
+    validationState,
+    validationText,
+    ...restProps
+  } = props;
+
+  const isDragAndDropSupported = 'draggable' in document.createElement('span');
+
+  const { isDragging, onChange, onDragOver, onDragEnter, onDragLeave, onDrop } = useFileUploaderInput({
+    accept,
+    maxFileSize,
+    maxUploadedFiles,
+    isMultiple,
+    onError,
+  });
+  const { classProps } = useFileUploaderStyleProps({
+    isDragAndDropSupported,
+    isDisabled,
+    isDragging,
+    isRequired,
+    validationState,
+  });
+  const { styleProps, props: transferProps } = useStyleProps(restProps);
+
+  const renderValidationText = useValidationText({
+    messageClassName: classProps.input.message,
+    validationState,
+    validationText,
+  });
+
+  return (
+    <div
+      {...transferProps}
+      {...styleProps}
+      onDragOver={isDragAndDropSupported ? onDragOver : undefined}
+      onDragEnter={isDragAndDropSupported ? onDragEnter : undefined}
+      onDragLeave={isDragAndDropSupported ? onDragLeave : undefined}
+      onDrop={isDragAndDropSupported ? onDrop : undefined}
+      className={classNames(classProps.input.root, styleProps.className)}
+    >
+      <label htmlFor={id} className={classProps.input.label}>
+        {label}
+      </label>
+      <input
+        type="file"
+        accept={accept}
+        id={id}
+        ref={inputRef}
+        className={classProps.input.input}
+        onChange={onChange}
+        multiple={isMultiple}
+        {...restProps}
+      />
+      <div ref={dropZoneRef} className={classProps.input.dropZone.root}>
+        <Icon name="upload" aria-hidden="true" />
+        <label htmlFor={id} className={classProps.input.dropZone.label}>
+          <span className={classProps.input.link}>{linkText}</span>
+          &nbsp;
```

Comment:
I think this better serves the purpose:

```suggestion
          {' '}
```

Reply:
I don't think it's a better way, I just prefer to put non-break space, because when the container is a flex-box, the space in the brackets usually disappears, but not the non-break space